src/features/game/expansion/components/UpcomingExpansion.tsx

```diff
@@ -142,21 +173,104 @@ export const UpcomingExpansion: React.FC = () => {
+              </Label>
+              <img
+                src={ITEM_DETAILS.Hammer.image}
+                className="w-20 mx-auto my-2"
+              />
+              <p className="text-sm text-center mb-2">
+                Visit the Fire Pit to cook food and feed your Bumpkin.
+              </p>
+            </div>
+          </CloseButtonPanel>
+        </Modal>
+
         <MapPlacement
           x={nextPosition.x - LAND_SIZE / 2}
           y={nextPosition.y + LAND_SIZE / 2}
           height={LAND_SIZE}
           width={LAND_SIZE}
         >
-          <div className="w-full h-full flex items-center justify-center opacity-90 hover:opacity-100">
+          <div className="w-full h-full flex flex-col items-center justify-center opacity-100">
             <img
               src={SUNNYSIDE.icons.expand}
               width={18 * PIXEL_SCALE}
-              className="relative cursor-pointer hover:img-highlight"
-              onClick={() => setShowBumpkinModal(true)}
+              className={classNames(
+                `relative cursor-pointer hover:img-highlight`,
+                {
+                  "animate-pulsate": showHelper,
+                }
+              )}
+              onClick={() => {
+                if (isLocked) {
+                  setShowLockedModal(true);
+                } else {
+                  setShowBumpkinModal(true);
+                }
+              }}
             />
+            {state.expansionRequirements && (
+              <>
+                <div className="flex mt-2">
+                  {getKeys(state.expansionRequirements?.resources ?? {}).map(
+                    (name) => (
+                      <div className="mr-3 flex items-center" key={name}>
+                        <RequirementLabel
+                          type="item"
+                          item={name}
+                          requirement={
+                            new Decimal(
+                              state.expansionRequirements?.resources[name] ?? 0
+                            )
+                          }
+                          balance={state.inventory[name] ?? new Decimal(0)}
+                        />
+                        {state.inventory[name]?.gte(
+                          state.expansionRequirements?.resources[name] ?? 0
+                        ) && (
+                          <img
+                            src={SUNNYSIDE.icons.confirm}
+                            className="h-4 ml-0.5"
+                          />
+                        )}
+                      </div>
+                    )
+                  )}
+                </div>
+                {isLocked && (
+                  <Label
+                    type="default"
+                    icon={lockIcon}
+                    className="mt-2"
+                  >{`Level ${state.expansionRequirements?.bumpkinLevel}`}</Label>
+                )}
+              </>
+            )}
```

Comment:
These appear to be flashing when I harvest crops - might need to abstract them into a separate component

Reply:
There are few known issues of flashing across the gameplay. Strangely some of these don't happen on mainnet.

I will check these out separately.

Reply:
No worries - I think it's likely the nested `Content` component. Not related to the changes in this PR. Just jotting down the [react/no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md#disallow-creating-unstable-components-inside-components-reactno-unstable-nested-components) rule. It may help us here.

```
"react/no-unstable-nested-components": ["error", { allowAsProps: true }],
```

Reply:
This is amazing! Nice find @c-py ⚡